sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivityViewModel.kt

```diff
@@ -128,64 +140,83 @@ class SyncActivityViewModel @Inject constructor(
 
     fun onToggleClicked(isChecked: Boolean) {
         viewModelScope.launch {
-            viewState.emit(viewState.value.toggle(isChecked))
+            viewState.value = viewState.value.toggle(isChecked)
             when (isChecked) {
                 true -> command.send(LaunchDeviceSetupFlow)
                 false -> {
                     syncRepository.getThisConnectedDevice()?.let {
                         command.send(AskTurnOffSync(it))
-                    } ?: hideOrShowAccountDetails()
+                    } ?: showAccountDetailsIfNeeded()
                 }
             }
         }
     }
 
     private suspend fun fetchRemoteDevices() {
-        val result = syncRepository.getConnectedDevices()
+        viewState.value = viewState.value.showDeviceListItemLoading()
+        val result = withContext(dispatchers.io()) {
+            syncRepository.getConnectedDevices()
+        }
         if (result is Success) {
             val newState = viewState.value.hideDeviceListItemLoading().setDevices(result.data.map { SyncedDevice(it) })
-            viewState.emit(newState)
+            viewState.value = newState
         } else {
-            viewState.emit(viewState.value.hideDeviceListItemLoading())
+            viewState.value = viewState.value.hideDeviceListItemLoading()
         }
     }
 
     fun onTurnOffSyncConfirmed(connectedDevice: ConnectedDevice) {
         viewModelScope.launch(dispatchers.io()) {
-            viewState.emit(viewState.value.hideAccount())
-            val result = syncRepository.logout(connectedDevice.deviceId)
-            if (result is Error) {
-                hideOrShowAccountDetails()
+            viewState.value = viewState.value.hideAccount()
+            when (syncRepository.logout(connectedDevice.deviceId)) {
+                is Error -> {
+                    viewState.value = viewState.value.toggle(true).showAccount()
+                }
+
+                is Success -> {
+                    viewState.value = signedOutState()
+                }
             }
         }
     }
 
     fun onTurnOffSyncCancelled() {
         viewModelScope.launch {
-            hideOrShowAccountDetails()
+            showAccountDetailsIfNeeded()
+        }
```

Comment:
Remove the `.launch`. `showAccountDetailsIfNeeded` is not a suspend fun

Reply:
done!